misc/json-to-record-converter/src/main/java/io/ballerina/jsonmapper/diagnostic/DiagnosticMessage.java

```diff
@@ -0,0 +1,86 @@
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package io.ballerina.jsonmapper.diagnostic;
+
+import io.ballerina.tools.diagnostics.DiagnosticSeverity;
+
+import java.util.Arrays;
+
+/**
+ * Contains diagnostic messages of the JSON to Record converter.
+ *
+ * @since 2201.2.0
+ */
+public final class DiagnosticMessage {
+
+    private final String code;
+    private final String description;
+    private final DiagnosticSeverity severity;
+    private final Object[] args;
+
+    private DiagnosticMessage(String code, String description, DiagnosticSeverity severity, Object[] args) {
+        this.code = code;
+        this.description = description;
+        this.severity = severity;
+        this.args = args;
+    }
+
+    public String getCode() {
+        return this.code;
+    }
+
+    public String getDescription() {
+        return this.description;
+    }
+
+    public DiagnosticSeverity getSeverity() {
+        return this.severity;
+    }
+
+    public Object[] getArgs() {
+        return this.args.clone();
+    }
+
+    public static DiagnosticMessage jsonToRecordConverter100(Object[] args) {
+        if (args != null) {
+            return new DiagnosticMessage("JSON_TO_RECORD_CONVERTER_100",
+                    String.format("Provided JSON is invalid : %s", DiagnosticUtils.
+                            transformJsonSyntaxErrorMessage((String) args[0])), DiagnosticSeverity.ERROR,
+                    Arrays.copyOfRange(args, 1, args.length));
+        }
+        return new DiagnosticMessage("JSON_TO_RECORD_CONVERTER_100",
+                 "Provided JSON is invalid.", DiagnosticSeverity.ERROR, null);
+    }
+
+    public static DiagnosticMessage jsonToRecordConverter101(Object[] args) {
+        return new DiagnosticMessage("JSON_TO_RECORD_CONVERTER_101",
+                "Provided JSON is unsupported. It may be null or have missing types.", DiagnosticSeverity.ERROR, args);
+    }
+
+    public static DiagnosticMessage jsonToRecordConverter102(Object[] args) {
+        return new DiagnosticMessage("JSON_TO_RECORD_CONVERTER_102",
+                "Error occurred while formatting the Ballerina syntax tree.", DiagnosticSeverity.ERROR, args);
+    }
+
+    public static DiagnosticMessage jsonToRecordConverter103(Object[] args) {
+        return new DiagnosticMessage("JSON_TO_RECORD_CONVERTER_103",
+                "There is no JSON entry be found for the field name %s. This is a serious issue.",
```

Comment:
Do we need this `This is a serious issue.` part since this is supposed to be a warning? 🙂 or am I missing something in here?

Reply:
As I explained, the code bloc which this message to be shown should not be reachable. Anyway since this is a warning, lemme refactor the error message.